src/arc.rs

```diff
@@ -144,6 +144,24 @@ impl<T> Arc<T> {
     }
 }
 
+impl<T> Arc<[T]> {
+    /// Reconstruct the `Arc<[T]>` from a raw pointer obtained from `into_raw()`.
+    ///
+    /// [`Arc::from_raw`] should accept unsized types, but this is not trivial to do correctly
+    /// until the feature [`pointer_bytes_offsets`](https://github.com/rust-lang/rust/issues/96283)
+    /// is stabilized. This is stopgap solution for slices.
+    pub unsafe fn from_raw_slice(ptr: *const [T]) -> Self {
+        let len = (*ptr).len();
+        // Assuming the offset of `T` in `ArcInner<T>` is the same
+        // as as offset of `[T]` in `ArcInner<[T]>`.
+        let arc_inner_ptr = (ptr as *const u8).sub(offset_of!(ArcInner<T>, data));
```

Comment:
question: is it not possible to just use `ArcInner<[T]>`?

Reply:
but the assumption is probably correct, at least due to CoerceUnsize behavior

Reply:
> question: is it not possible to just use ArcInner<[T]>?

`offset_of!` macro doesn't work with unsized types.